scripts/update-corlib-version

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/perl
+
+use warnings FATAL => 'all';
```

Comment:
Use bash.

Reply:
Bash is to be avoided imho. Python is good choice though.

Reply:
Every time I try to do anything in Bourne shell, many times across many years, I always come away disappointed and resolved not to waste more time on it.
It feels like a pile of nonsense and not an actually designed language.
This time the same.

For example:

```
!/usr/bin/bash

uuid=`uuidgen`

while IFS= read -r line; do
	printf "%s\n" "${line//CORLIB_VERSION=.+/CORLIB_VERSION=$uuid}"
done
```

Close, but the .+ part of the regular expression doesn't work.

It is noticably slower than Perl, probably due to applying the regexp on every line instead of en masse once over the entire file.

Attempt to run regexp just once is faster when the regexp doesn't match but hangs and cannot control-c or z. Perhaps it kinda loops.

```
#!/bin/bash

contents=`cat configure.ac`

uuid=`uuidgen`

printf "%s\n" "${contents//CORLIB_VERSION=/CORLIB_VERSION=$uuid}"
```

Perhaps we can agree on Python?
But the Python 2.x vs. 3.x situation suggests Python developers do not take compatibility seriously enough.

Anyway, if Perl isn't acceptable, I wrote it in C also. That is easier and more portable than bash.

Reply:
We definitely don't want a C version and even the bash version is debatable. 

I don't think we need a helper script for essentially running `uuidgen` and copy/pasting a string to a file, you can do that manually in 2 seconds.

Add a comment to configure.ac instead, something along the lines of:

```
# Use the uuidgen command to create a unique string
MONO_CORLIB_VERSION=07F6F369B78D4D709FE90C0EA002A970
```

Reply:
I think command + copy/paste is a bit much to ask, when the automation is so easy. As an obvious side-benefit, the automation is now easy. It was kinda impossible before, because I would do such things as try to increment past concurrent PRs.

Reply:
The C version is because Perl was objected to. I agree it is borderline. However, consider that you might want this is part of a nightly or weekly automation as well. i.e. 1. Otherwise we are promulgating binaries, which while built by infrastructure, derive from temporary private branches, not actual current merged master 2. Because the behavioral vs. calling interface variable is perhaps subtle and people won't abide by it, so rebuild every so often just in case.

Reply:
I wrong about infra needing to increment/change. Like today, it can rebuild and overwrite.

Reply:
Bash doesn't introduce a dependency. We don't want more dependencies than we actually need. Pretty easy to apply, especially in this case where all we want to do is invoke an executable. And as pointed by @akoeplinger, even easier: don't have a script but document what's needed.

Reply:
The dependency is optional as well.